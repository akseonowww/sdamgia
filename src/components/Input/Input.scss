@import 'src/variables';

.Input {
	box-sizing: border-box;
	padding: 0.3rem 0.55rem 0.4rem;

	font-family: inherit;
	font-size: inherit;

	background: $input-bg;
	border: none;
	border-radius: $border-radius-input;
	outline: none;
	box-shadow: none;
}
