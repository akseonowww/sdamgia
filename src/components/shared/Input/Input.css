.Input {
	box-sizing: border-box;
	margin: 0;
	padding: 0.3rem 0.55rem 0.4rem;

	font-size: 1rem;

	background: #f0f0f0;
	border: none;
	border-radius: 3px;
	box-shadow: none;
	outline: none;
}
