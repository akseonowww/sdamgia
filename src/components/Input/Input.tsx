import React, {
  FC,
  useCallback,
  Dispatch,
  SetStateAction,
  MouseEvent,
  ChangeEvent,
  CSSProperties,
  RefObject,
} from 'react'
import cx from 'classnames'

import './Input.scss'

interface IInputProps {
  className?: string
  type?: string
  name: string
  value: string
  placeholder?: string
  autoFocus?: boolean
  tabIndex?: number
  style?: CSSProperties
  innerRef?: RefObject<HTMLInputElement>
  setValue?: Dispatch<SetStateAction<string>>
  handleClick?: (e: MouseEvent<HTMLInputElement>) => void
  handleChange?: (e: ChangeEvent<HTMLInputElement>) => void
}

const Input: FC<IInputProps> = ({
  className,
  type = 'text',
  name,
  value,
  placeholder,
  autoFocus,
  tabIndex,
  style,
  innerRef,
  setValue,
  handleClick: handleSpecialClick,
  handleChange: handleSpecialChange,
}) => {
  const handleChange = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      if (!setValue) return

      setValue(e.target.value)
    },
    [setValue]
  )

  return (
    <input
      className={cx('Input', className)}
      type={type}
      name={name}
      value={value}
      placeholder={placeholder}
      autoFocus={autoFocus}
      tabIndex={tabIndex}
      style={style}
      ref={innerRef}
      onClick={handleSpecialClick}
      onChange={handleSpecialChange ? handleSpecialChange : handleChange}
      data-testid="Input"
    />
  )
}

export default Input
