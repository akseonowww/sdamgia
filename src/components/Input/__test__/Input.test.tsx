import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import Input from '..'

describe('Input', () => {
  let container: HTMLElement | null = null
  const setValue = jest.fn(() => console.log('To change state value'))

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('should render a content', () => {
    act(() => {
      render(
        <Input
          className="Form_oneLine-Input"
          type="tel"
          name="id"
          value="0"
          placeholder="Номер варианта"
          setValue={setValue}
        />,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
