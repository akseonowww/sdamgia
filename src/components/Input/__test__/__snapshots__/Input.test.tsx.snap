// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render a content 1`] = `"<input class=\\"Input Form_oneLine-Input\\" type=\\"tel\\" name=\\"id\\" placeholder=\\"Номер варианта\\" value=\\"0\\">"`;
